rest-api/dao/physical_measurements_dao.py

```diff
@@ -291,28 +291,45 @@ def _update_participant_summary(self, session, obj):
       raise BadRequest("Can't submit physical measurements for participant %s without consent" %
                        participant_id)
     raise_if_withdrawn(participant_summary)
-    participant_summary.physicalMeasurementsTime = obj.created
-    participant_summary.physicalMeasurementsFinalizedTime = obj.finalized
-    participant_summary.physicalMeasurementsCreatedSiteId = obj.createdSiteId
-    participant_summary.physicalMeasurementsFinalizedSiteId = obj.finalizedSiteId
     participant_summary.lastModified = clock.CLOCK.now()
 
-    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED:
-      if not self.has_uncancelled_pm(session, participant):
-        # update corresponding participant summary physicalMeasurementsStatus
-        participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
-        participant_summary.physicalMeasurementsTime = None
+    # These fields set on measurement that is cancelled and doesn't have a previous good measurement
+    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and not \
+       self.has_uncancelled_pm(session, participant):
+      participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.CANCELLED
+      participant_summary.physicalMeasurementsTime = None
 
-    if participant_summary.physicalMeasurementsStatus != PhysicalMeasurementsStatus.CANCELLED or \
-       obj.status != PhysicalMeasurementsStatus.CANCELLED:
-      # if a PM was restored, it is complete again.
+    # These fields set on any measurement not cancelled
+    elif obj.status != PhysicalMeasurementsStatus.CANCELLED:
+      # new PM or if a PM was restored, it is complete again.
       participant_summary.physicalMeasurementsStatus = PhysicalMeasurementsStatus.COMPLETED
+      participant_summary.physicalMeasurementsTime = obj.created
+      participant_summary.physicalMeasurementsFinalizedTime = obj.finalized
+      participant_summary.physicalMeasurementsCreatedSiteId = obj.createdSiteId
+      participant_summary.physicalMeasurementsFinalizedSiteId = obj.finalizedSiteId
+
+    if obj.status and obj.status == PhysicalMeasurementsStatus.CANCELLED and \
+       self.has_uncancelled_pm(session, participant):
+
+      get_latest_pm = self.get_latest_pm(session, participant)
+      participant_summary.physicalMeasurementsFinalizedTime = get_latest_pm.finalized
+      participant_summary.physicalMeasurementsTime = get_latest_pm.created
+      participant_summary.physicalMeasurementsCreatedSiteId = get_latest_pm.createdSiteId
+      participant_summary.physicalMeasurementsFinalizedSiteId = get_latest_pm.finalizedSiteId
+
 
     participant_summary_dao.update_enrollment_status(participant_summary)
     session.merge(participant_summary)
 
     return participant_summary
 
+  def get_latest_pm(self, session, participant):
+    query = session.query(PhysicalMeasurements).filter(PhysicalMeasurements.status.isnot(
```

Comment:
nit: I'm not a SQLAlchemy expert, but maybe put the participant filter first? It probably won't matter but I'm slightly scared this would do something like pull all uncancelled PM into memory first.

Reply:
done